### DISCUSSION TITLE
Error page in hybrid app with SPA and SSG

### DISCUSSION THREAD

**Discussion by truuman** [vikejs/vike#699] [Discussion]
This is a Vue project. This is an overview of the project structure:

```
src/admin/renderer/app.ts
src/admin/renderer/PageShell.vue
src/admin/renderer/_default.page.server.ts
src/admin/renderer/_default.page.client.ts
src/main/renderer/app.ts
src/main/renderer/PageShell.vue
src/main/renderer/_default.page.server.ts
src/main/renderer/_default.page.client.ts
src/main/_default.page.route.ts
src/renderer/_error.page.vue
```

I have multiple [multiple defaults for page rendering](https://vite-plugin-ssr.com/default-page#multiple-defaults) and my `pages` are set up with two domains, `main` and `admin`.

`main` relies entirely on SSG, it has its own `renderer` folder with its own `app.ts`, `PageShell.vue`, `_default.page.server.ts` and `_default.page.client.ts`. It also has, in the `main` root a `_default.page.route.ts` to utilize [fileSystemRoutingRoot](https://vite-plugin-ssr.com/filesystemRoutingRoot).

`admin` relies entirely on SPA, it has its own `renderer` folder with its own `app.ts`, `PageShell.vue`, `_default.page.server.ts` and `_default.page.client.ts`.

Having an [_error.page.vue](https://vite-plugin-ssr.com/error-page) in the shared `renderer` folder causes the application to always load the `admin` renderer when there is an error - meaning the error page is in SPA. If I place `_error.page.vue` in the `src/main/renderer` folder, then it's the SSR/prerendered error page that loads, even on the admin page.

Is this the expected behavior? Wouldn't it make sense in this case to have error pages per domain?

**Comment by brillout** [comment]
Which `_error.page.vue` should be picked if the user goes to `/some-404-url`? That's the reason why VPS only allows a single error page.

**Reply by truuman** [reply]
Would be nice if I could have a default, and then override as required. Like in the case of authenticated users who might be seeing a different UI.

**Reply by brillout** [reply]
I did consider this but I couldn't find a satisfactory design for it. Feel free to make a design proposal.
